### PR TITLE
[AdminListBundle]: Add Form & Request property and Response function in AdminListEvent

### PR DESCRIPTION
**Pull request by mattiasdelang**
| Q             | A
| ------------- | ---
| Bug fix?      |no
| New feature?  | yes
| BC breaks?   | yes
| Deprecations?|no

Add Form and Request property in the AdminListEvent to have more flexibility in adapting the adminlistscontroller.
Also added a Response function, so you can give a custom Response in the adminlistcontoller.

### CODE REVIEW

File: src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php [hunk 1]
```diff
@@ -12,12 +14,26 @@ class AdminListEvent extends Event
...
+    protected $request;
+
+    /**
      * AdminListEvent constructor.
      * @param object $entity
+     * @param FormInterface $form
+     * @param Request $request
      */
-    public function __construct($entity)
+    public function __construct($entity, Request $request, FormInterface $form = null)
```
Comment (aistis-) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: It is a BC, please mention it in the UPGRADE :)

File: src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php [hunk 2]
```diff
@@ -12,12 +14,26 @@ class AdminListEvent extends Event
...
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
      * AdminListEvent constructor.
      * @param object $entity
+     * @param FormInterface $form
```
Comment (aistis-) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: Params switched the places. Also `@param FormInterface|null $form`

File: src/Kunstmaan/AdminListBundle/Controller/AdminListController.php [hunk 3]
```diff
@@ -401,19 +425,19 @@ protected function buildSortableFieldActions(AbstractAdminListConfigurator $conf
...
             };
 
             $action = new SimpleItemAction($route, 'arrow-up', 'Move up');
             $configurator->addItemAction($action);
 
-            $route = function (EntityInterface $item) use ($configurator){
+            $route = function (EntityInterface $item) use ($configurator) {
                 return array(
-                    'path' => $configurator->getPathByConvention() . '_move_down',
+                    'path' => $configurator->getPathByConvention().'_move_down',
```
Comment (aistis-) [inline src/Kunstmaan/AdminListBundle/Controller/AdminListController.php]: why removing the spaces?

File: src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php [hunk 4]
```diff
@@ -27,4 +43,21 @@ public function getEntity()
...
+    /**
+     * @return Request
+     */
+    public function getRequest()
+    {
+        return $this->request;
+    }
+
+    /**
+     * @return FormInterface
```
Comment (aistis-) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: `FormInterface|null`

File: src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php (hunk 1 above)
Comment (mattiasdelang) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: ill make them both nullable

Comment (aistis-) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: Personally, I do not like `nullable` if it is not necessary :) it depends on case. We can allow BC I guess, as we do not have stable version for v4

@jockri

Comment (mattiasdelang) [inline src/Kunstmaan/AdminListBundle/Event/AdminListEvent.php]: Made it a BC and is mentioned in the UPGRADE

**Pull request merged**